p2p/p2p.go

```diff
@@ -19,77 +19,92 @@ package p2p
 import (
 	"bufio"
 	"context"
-	"fmt"
+	"time"
 
-	module "github.com/ChainSafe/gossamer/internal/api/modules"
+	"github.com/ChainSafe/gossamer/common"
 	"github.com/ChainSafe/gossamer/internal/services"
 	log "github.com/ChainSafe/log15"
 
+	"github.com/libp2p/go-libp2p-core/network"
 	net "github.com/libp2p/go-libp2p-core/network"
 )
 
 var _ services.Service = &Service{}
 
-// Service describes a p2p service, including host and dht
+// SendStatusInterval is the time between broadcasting status messages
+const SendStatusInterval = 5 * time.Minute
+
+// Service describes a p2p service
 type Service struct {
-	ctx              context.Context
-	host             *host
-	msgSend          chan<- Message
-	msgRec           <-chan Message
-	blockReqRec      map[string]bool
-	blockRespRec     map[string]bool
-	blockAnnounceRec map[string]bool
-	txMessageRec     map[string]bool
+	ctx          context.Context
+	host         *host
+	msgRec       <-chan Message
+	msgSend      chan<- Message
+	blockAnnRec  map[string]bool
+	blockReqRec  map[string]bool
+	blockRespRec map[string]bool
+	txMessageRec map[string]bool
+}
+
+// TODO: use generated status message
+var statusMessage = &StatusMessage{
+	ProtocolVersion:     0,
+	MinSupportedVersion: 0,
+	Roles:               0,
+	BestBlockNumber:     0,
+	BestBlockHash:       common.Hash{0x00},
+	GenesisHash:         common.Hash{0x00},
+	ChainStatus:         []byte{0},
 }
 
-// NewService creates a new p2p.Service using the service config. It initializes the host and dht
+// NewService creates a new p2p service using the provided configuration and message channels
 func NewService(conf *Config, msgSend chan<- Message, msgRec <-chan Message) (*Service, error) {
 	ctx := context.Background()
 
-	host, err := newHost(ctx, conf)
+	h, err := newHost(ctx, conf)
 	if err != nil {
 		return nil, err
 	}
 
 	s := &Service{
-		ctx:     ctx,
-		host:    host,
-		msgSend: msgSend,
-		msgRec:  msgRec,
+		ctx:          ctx,
+		host:         h,
+		msgRec:       msgRec,
+		msgSend:      msgSend,
+		blockAnnRec:  make(map[string]bool),
+		blockReqRec:  make(map[string]bool),
+		blockRespRec: make(map[string]bool),
+		txMessageRec: make(map[string]bool),
 	}
 
-	host.registerStreamHandler(s.handleStream)
-
-	s.blockReqRec = make(map[string]bool)
-	s.blockRespRec = make(map[string]bool)
-	s.blockAnnounceRec = make(map[string]bool)
-	s.txMessageRec = make(map[string]bool)
+	h.registerStreamHandler(s.handleStream)
 
 	return s, err
 }
 
-// Start begins the p2p Service, including discovery
+// Start starts the service
 func (s *Service) Start() error {
 	s.host.startMdns()
 	s.host.bootstrap()
-	s.host.logAddrs()
+	s.host.printHostAddresses()
 
-	log.Info("Listening for connections...")
+	// TODO: decide whether we need errors to surface here
+	// e := make(chan error)
```

Comment:
Worth pointing out. David mentioned the error channel might not be necessary here.